File: Main/ArchiMeter.Analysis/VariableNameProvider.fs
namespace ArchiMeter.Analysis.Xaml
open System.Collections.Generic

type internal VariableNameProvider() =
    static let UnnamedCache = new Dictionary<string, int>()
    static member GetUnnamedVariable(className : string) =
        if not (UnnamedCache.ContainsKey className) then
            UnnamedCache.[className] <- 0
        let number = UnnamedCache.[className]
        let name = className.ToLower() + number.ToString()
        UnnamedCache.[className] <- (number + 1)
        name.Replace(".", "_")

File: ArchiMetrics.Common/Xaml/VariableNameProvider.cs
namespace ArchiMetrics.Common.Xaml
{
	using System.Collections.Generic;

	internal class VariableNameProvider
	{
		private static readonly Dictionary<string, int> UnnamedCache = new Dictionary<string, int>();

		public static string Get(string className)
		{
			if (UnnamedCache.ContainsKey(className) == false)
			{
				UnnamedCache.Add(className, 0);
			}

			var name = className.ToLower() + UnnamedCache[className];

			UnnamedCache[className]++;

			return name.Replace(".", "_");
		}
	}
}
